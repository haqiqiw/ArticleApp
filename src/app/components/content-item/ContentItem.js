import React from 'react';
import {
  ListItem,
  Thumbnail,
  Body,
  Text
} from 'native-base';

const ContentItem = ({
  onPressItem, 
  thumbnail,
  thumbnailSize,
  thumbnailStyle,
  title,
  titleStyle,
  titleLine,
  subtitle,
  subtitleStyle,
  subtitleLine
}) => {
  return (
    <ListItem button onPress={onPressItem}>
      { thumbnail && <Thumbnail style={thumbnailStyle} square size={thumbnailSize} source={thumbnail} /> }
      <Body>
        <Text style={titleStyle} numberOfLines={titleLine}>{title}</Text>
        <Text style={subtitleStyle} numberOfLines={subtitleLine} note>{subtitle}</Text>
      </Body>
    </ListItem>
  );
}

ContentItem.defaultProps = {
  onPressItem: () => null,
  thumbnailSize: 80
}

export default ContentItem;