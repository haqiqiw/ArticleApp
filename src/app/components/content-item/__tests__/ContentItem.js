import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ContentItem from '../ContentItem';

describe('ContentItem Component', () => {
  it('should render without issue', () => {
    const component = shallow(<ContentItem />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should pass title props should still work', () => {
    const component = shallow(<ContentItem title="Title Content" />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should apply title style in the contentitem', () => {
    const component = shallow(
      <ContentItem titleStyle={{ fontSize: 18 }} />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should pass subtitle props should still work', () => {
    const component = shallow(<ContentItem subtitle="Subtitle Content" />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should apply subtitle style in the contentitem', () => {
    const component = shallow(
      <ContentItem subtitleStyle={{ fontSize: 16 }} />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});