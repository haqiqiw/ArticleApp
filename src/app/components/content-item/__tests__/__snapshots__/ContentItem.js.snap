// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentItem Component should apply subtitle style in the contentitem 1`] = `
<Styled(ListItem)
  button={true}
  onPress={[Function]}
>
  <Styled(Body)>
    <Styled(Text) />
    <Styled(Text)
      note={true}
      style={
        Object {
          "fontSize": 16,
        }
      }
    />
  </Styled(Body)>
</Styled(ListItem)>
`;

exports[`ContentItem Component should apply title style in the contentitem 1`] = `
<Styled(ListItem)
  button={true}
  onPress={[Function]}
>
  <Styled(Body)>
    <Styled(Text)
      style={
        Object {
          "fontSize": 18,
        }
      }
    />
    <Styled(Text)
      note={true}
    />
  </Styled(Body)>
</Styled(ListItem)>
`;

exports[`ContentItem Component should pass subtitle props should still work 1`] = `
<Styled(ListItem)
  button={true}
  onPress={[Function]}
>
  <Styled(Body)>
    <Styled(Text) />
    <Styled(Text)
      note={true}
    >
      Subtitle Content
    </Styled(Text)>
  </Styled(Body)>
</Styled(ListItem)>
`;

exports[`ContentItem Component should pass title props should still work 1`] = `
<Styled(ListItem)
  button={true}
  onPress={[Function]}
>
  <Styled(Body)>
    <Styled(Text)>
      Title Content
    </Styled(Text)>
    <Styled(Text)
      note={true}
    />
  </Styled(Body)>
</Styled(ListItem)>
`;

exports[`ContentItem Component should render without issue 1`] = `
<Styled(ListItem)
  button={true}
  onPress={[Function]}
>
  <Styled(Body)>
    <Styled(Text) />
    <Styled(Text)
      note={true}
    />
  </Styled(Body)>
</Styled(ListItem)>
`;
